control/src/Controller/CommonSMC.cpp

```diff
@@ -156,7 +156,6 @@ void CommonSMC::initialize(const Model& m)
   }
 
   _SMC->nonSmoothDynamicalSystem()->insertDynamicalSystem(_DS_SMC);
-  _SMC->nonSmoothDynamicalSystem()->topology()->setOSI(_DS_SMC, _integratorSMC);
```

Comment:
Oh right, but then this imposes an ugly and useless need to create the Simulation object before adding DSs to the Model, which is not how many of the examples work.  I'll do it anyways for now just to see how it looks, but it seems like a wrong turn.  Otherwise we go back to calling topology()- setOSI() manually, but then have two distinct cases:

1. DS is created before the Simulation, call insertDynamicalSystem and nsds->topology()->setOSI
2. DS is created after the Simulation, call insertDynamicalSystem and Simulation::initializeOSIforDS.

Confusing.